backend/test/unit/controller/startController.test.ts

```diff
@@ -0,0 +1,126 @@
+import { expect, test, jest, afterEach } from '@jest/globals';
+import { Response } from 'express';
+
+import { handleStart } from '@src/controller/startController';
+import { StartGetRequest } from '@src/models/api/StartGetRequest';
+import { LEVEL_NAMES } from '@src/models/level';
+
+jest.mock('@src/promptTemplates', () => ({
+	systemRoleLevel1: 'systemRoleLevel1',
+	systemRoleLevel2: 'systemRoleLevel2',
+	systemRoleLevel3: 'systemRoleLevel3',
+}));
+
+const mockSend = jest.fn();
+
+function responseMock() {
+	return {
+		send: mockSend,
+		status: jest.fn().mockReturnValue({ send: mockSend }),
+	} as unknown as Response;
+}
+
+afterEach(() => {
+	mockSend.mockClear();
+});
+
+[
+	LEVEL_NAMES.LEVEL_1,
+	LEVEL_NAMES.LEVEL_2,
+	LEVEL_NAMES.LEVEL_3,
+	LEVEL_NAMES.SANDBOX,
+].forEach((level) => {
+	test(`GIVEN level ${
+		level + 1
+	} provided WHEN user starts the frontend THEN the backend sends the correct initial information`, () => {
+		const req = {
+			query: {
+				level,
+			},
+			session: {
+				levelState: [
+					{
+						sentEmails: 'level 1 emails',
+						chatHistory: 'level 1 chat history',
+						defences: 'level 1 defences',
+					},
+					{
+						sentEmails: 'level 2 emails',
+						chatHistory: 'level 2 chat history',
+						defences: 'level 2 defences',
+					},
+					{
+						sentEmails: 'level 3 emails',
+						chatHistory: 'level 3 chat history',
+						defences: 'level 3 defences',
+					},
+					{
+						sentEmails: 'level 4 emails',
+						chatHistory: 'level 4 chat history',
+						defences: 'level 4 defences',
+					},
+				],
+				systemRoles: [],
+			},
+		} as unknown as StartGetRequest;
+		const res = responseMock();
+
+		handleStart(req, res);
+
+		expect(mockSend).toHaveBeenCalledWith({
+			emails: `level ${level + 1} emails`,
```

Comment:
Not certain what you're referring to. I'm going to assume it's the fact that we're setting and returning string values in our mocked session rather than the types we have set elsewhere.

Isn't it a good thing for tests to be able to ignore the complex types of the real thing, so that the tests can be kept as simple as possible?

Anyway, this is only allowed because of the type casting: `const req = {...} as unknown as StartGetRequest`. We're explicitly and intentionally going over the type system.

Reply:
Yes that's what I mean. We are verifying that a function was called with something it will (and can) never be called with in the actual implementation. Just looked odd that's all, doesn't really matter as we're only using those "somethings" to verify we were using the correct level.